src/main/java/org/broadinstitute/hellbender/engine/PathSpecifier.java

```diff
@@ -0,0 +1,260 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.*;
+import java.nio.file.spi.FileSystemProvider;
+
+/**
+ * Default implementation for PathURI.
+ *
+ * This class takes a raw string that is to be interpreted as a path specifier, and converts it internally to a
+ * URI and/or Path object. If no scheme is provided as part of the raw string used in the constructor(s), the
+ * input is assumed to represent a file on the local file system, and will be backed by a URI with a "file:/"
+ * scheme and a path part that is automatically encoded/escaped to ensure it is a valid URI. If the raw string
+ * contains a scheme, it will be backed by a URI formed from the raw string as presented, with no further
+ * encoding/escaping.
+ *
+ * For example, a URI that contains a scheme and has an embedded "#" in the path will be treated as a URI
+ * having a fragment delimiter. If the URI contains an scheme, the "#" will be escaped and the encoded "#"
+ * will be interpreted as part of the URI path.
+ *
+ * There are 3 succeeding levels of input validation/conversion:
+ *
+ * 1) PathSpecifier constructor: requires a syntactically valid URI, possibly containing a scheme (if no scheme
+ *    is present the path part will be escaped/encoded), or a valid local file reference
+ * 2) isNio: true if the input string is an identifier that is syntactically valid, and is backed by
```

Comment:
What do you think about getting rid of this completely for now ? Most of the cases that I know of will branch based on the scheme, otherwise they assume its Path-able. I'm still not sure there is a use-case where something will call this and do something useful when it returns false.

Reply:
We could remove it... I'm also not totally clear on the use case.  Maybe for displaying an error message?